### PR TITLE
azurerm_databricks_workspace fix for #2214

### LINKED ISSUE #2214: Unable to launch workspace after create succesfully an azurerm_databricks_workspace
Using the new "azurerm_databricks_workspace" i'm unable to launch the workspace. It's created and accesible through Azure Portal, but when I try to "Launch Workspace" it is unable to access (the error code is that the workspace doesn't exists on that location)

I was doing some test using an ARM template (trying to replicate the same error than the databricks created with terraform) and I think that the problem is the uppercase in SKU. If I create a databricks with my ARM template and select SKU to "Standard" (Uppercase) it doesn't work. But If launch the same ARM template with SKU to "standard" (Lowercase) it's fully accesible.

It could be an error or maybe I'm doing something wrong?

Thank you so much. Regards!

### PR DESCRIPTION
**Pull request by tiwood**
This fixes #2214 and adds an additional (optional) parameter to define the name of the resource group which contains the managed Databricks resources.

```hcl
resource "azurerm_resource_group" "test" {
  name     = "example-resources"
  location = "West US"
}

resource "azurerm_databricks_workspace" "test" {
  name                  = "databricks-test"
  resource_group_name   = "${azurerm_resource_group.test.name}"
  location              = "${azurerm_resource_group.test.location}"
  sku                   = "Standard"
  managed_resource_group = "managed-databricks-resources"

  tags {
    Environment = "Production"
  }
}
```

### CODE REVIEW

File: azurerm/resource_arm_databricks_workspace.go [hunk 1]
```diff
@@ -39,13 +39,21 @@ func resourceArmDatabricksWorkspace() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 				ValidateFunc: validation.StringInSlice([]string{
-					"Standard",
-					"Premium",
+					"standard",
+					"premium",
```
[the remark this answers is not in the document]
Comment (tiwood) [reply azurerm/resource_arm_databricks_workspace.go]: Changing the SKU to lower-case did it for me. I could successfully login to the workspace right after the deployment completed.

[the remark this answers is not in the document]
Comment (tiwood) [reply azurerm/resource_arm_databricks_workspace.go]: It seems the Databricks workspace does some auto configuration as soon as the first user tries to sign in. This exact process fails if the SKU is not lower-case.

The ARM deployment will complete successfully even if we use 'Standard' or 'Premium".

File: azurerm/resource_arm_databricks_workspace_test.go [hunk 2]
```diff
@@ -194,48 +194,50 @@ resource "azurerm_databricks_workspace" "test" {
...
 resource "azurerm_databricks_workspace" "test" {
-  name                = "acctestdbw-%d"
-  resource_group_name = "${azurerm_resource_group.test.name}"
-  location            = "${azurerm_resource_group.test.location}"
-  sku                 = "Standard"
+  name                		= "acctestdbw-%d"
+  resource_group_name		 	= "${azurerm_resource_group.test.name}"
+  location            		= "${azurerm_resource_group.test.location}"
+	sku                 		= "standard"
+	managed_resource_group 	= "acctestRG-%d-managed"
```
[the remark this answers is not in the document]
Comment (tiwood) [reply azurerm/resource_arm_databricks_workspace_test.go]: Fixed!

File: azurerm/resource_arm_databricks_workspace_test.go [hunk 3]
```diff
@@ -194,48 +194,50 @@ resource "azurerm_databricks_workspace" "test" {
...
 resource "azurerm_databricks_workspace" "test" {
-  name                = "acctestdbw-%d"
-  resource_group_name = "${azurerm_resource_group.test.name}"
-  location            = "${azurerm_resource_group.test.location}"
-  sku                 = "Standard"
+  name                		= "acctestdbw-%d"
+  resource_group_name 		= "${azurerm_resource_group.test.name}"
+  location            		= "${azurerm_resource_group.test.location}"
+	sku                 		= "standard"
+	managed_resource_group 	= "acctestRG-%d-managed"
```
[the remark this answers is not in the document]
Comment (tiwood) [reply azurerm/resource_arm_databricks_workspace_test.go]: Fixed!

File: website/docs/r/databricks_workspace.html.markdown [hunk 4]
```diff
@@ -42,6 +42,10 @@ The following arguments are supported:
 
 * `sku` - (Required) The `sku` to use for the Databricks Workspace. Possible values are `Standard` or `Premium`. Changing this forces a new resource to be created.
 
+* `managed_resource_group` - (Optional) The name of the resource group where Azure should place the managed Databricks resources. Changing this forces a new resource to be created.
+
+~> **NOTE** You have to provide a unused resource group name in `managed_resource_group` as Azure has to create the resource group as 'managed' resource group during the deployment. The deployment will fail if a resource group with the same name in the same subscription already exists.
```
[the remark this answers is not in the document]
Comment (tiwood) [reply website/docs/r/databricks_workspace.html.markdown]: Agreed and changed!

**Comment by tiwood** [comment]
Thanks @tombuildsstuff, I've applied the changes as suggested.

**Comment by katbyte** [comment]
Just ran the tests for this, import fails as you are not reading & setting the name of the managed resource group back.

```
------- Stdout: -------
=== RUN   TestAccAzureRMDatabricksWorkspace_complete
=== PAUSE TestAccAzureRMDatabricksWorkspace_complete
=== CONT  TestAccAzureRMDatabricksWorkspace_complete
--- FAIL: TestAccAzureRMDatabricksWorkspace_complete (708.91s)
	testing.go:538: Step 2 error: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
		
		(map[string]string) {
		}
		
		
		(map[string]string) (len=1) {
		 (string) (len=22) "managed_resource_group": (string) (len=37) "acctestRG-1837937184875173389-managed"
		}
		
FAIL
```

File: azurerm/resource_arm_databricks_workspace.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply azurerm/resource_arm_databricks_workspace.go]: 👍 sounds good - I've opened an upstream bug about this to try and get the underlying bug fixed: https://github.com/Azure/azure-rest-api-specs/issues/4732

**Comment by tiwood** [comment]
Hi @katbyte, should be fixed. Tests pass.

```
=== RUN   TestAccAzureRMDatabricksWorkspace_complete
=== PAUSE TestAccAzureRMDatabricksWorkspace_complete
=== CONT  TestAccAzureRMDatabricksWorkspace_complete
--- PASS: TestAccAzureRMDatabricksWorkspace_complete (697.86s)
PASS
```

**Pull request merged**